packages/go_router/lib/src/go_router_delegate.dart

```diff
@@ -715,20 +216,24 @@ class GoRouterDelegate extends RouterDelegate<Uri>
 
       // get a page from the builder and associate it with a sub-location
       final GoRouterState state = GoRouterState(
-        this,
-        location: location,
+        _parser,
+        location: match.fullUriString,
```

Comment:
One small question: What is the motivation behind this change?

Before go_router 4 I received the full location which was requested, now I only get the location of the sub route.

So If I have the following routes:

  => /family
  =>   /family/:pid
  =>     /family/:pid/info
  
And I opt to open the following location "/family/123/info" I will get for every builder call:

 /family => state.location = "/family"
 /family/:pid=> state.location = "/family/123"
 /family/:pid/info => state.location = "/family/123/info"

but before the change I got:

 /family => state.location = "/family/123/info"
 /family/:pid=> state.location = "/family/123/info"
 /family/:pid/info => state.location = "/family/123/info"

Reply:
Hi @rzetzsche, could you file a new issue for this? This was probably an oversight.

Reply:
Thank you!